src/sync.cpp

```diff
@@ -3040,6 +3058,43 @@ error JSONSyncConfigIOContext::read(const LocalPath& dbPath,
     return API_OK;
 }
 
+error JSONSyncConfigIOContext::remove(const LocalPath& dbPath,
+                                      const unsigned int slot)
+{
+    using std::to_string;
+
+    LocalPath path = dbPath;
+
+    // Generate the rest of the path.
+    path.appendWithSeparator(mName, false);
+    path.append(LocalPath::fromPath("." + to_string(slot), mFsAccess));
+
+    return mFsAccess.unlinklocal(path) ? API_OK : API_EWRITE;
+}
+
+error JSONSyncConfigIOContext::remove(const LocalPath& dbPath)
+{
+    vector<unsigned int> confSlots;
+
+    // What slots are present on disk?
+    if (getSlotsInOrder(dbPath, confSlots) == API_ENOENT)
+    {
+        // None so nothing to do.
+        return API_ENOENT;
+    }
+
+    bool result = true;
+
+    // Remove the slots from disk.
+    for (auto confSlot : confSlots)
+    {
+        result &= remove(dbPath, confSlot) == API_OK;
```

Comment:
Same thing here. Would an error message help debugging in case of failure?

Reply:
I haven't added any extra logging regarding this function.

Logging is now done above and below.

That is, truncate(...) warns if this function fails.
And remove(...) logs if it cannot remove the specified slot.